.github/workflows/calibreapp-image-action.yml

```
@@ -4,35 +4,67 @@
 # 
 # This runs the calibre GitHub action to compress our images
 #
-# This Action has a few limitations:
-#
-# 1) This currently only works on pull requests from this repo
-#    and fails silently for PRs from forks. We've added a PR
-#    to handle this better (https://github.com/calibreapp/image-actions/pull/54)
-#    and after that can probably combine to the generate_chapters.yml.
-#
-# 2) It seems to only do a basic level of optimisations and better results can
-#    be achieved from https://tinypng.com/ or https://squoosh.app/ but not as automated
-#    so still good to have this.
-#
-# 3) Currently doesn't handle WebP (https://github.com/calibreapp/image-actions/issues/31)
+# This Action has a few different methoids for running:
+# 1) For Pull Requests from and to the main repo it can run and update the PR
+# 2) For pushes to main, it will run and create a new PR. This captures PR from
+#    Forks.
+# 3) It also runs every sunday at 11pm in case anything else sneaks in or any
+#    improvements are added to Calibre. This will also open a PR if necessary.
+# 4) Finally it can be kicked off manually and it will open a PR if necessary.
 #
 name: Compress images
 on:
+  schedule:
+    # Run a full compression every Sunday at 11pm to start the week right
+    - cron: '00 23 * * 0'
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - '**.jpg'
+      - '**.jpeg'
+      - '**.png'
+      - '**.webp'
   pull_request:
     paths:
       - '**.jpg'
+      - '**.jpeg'
       - '**.png'
       - '**.webp'
 jobs:
   build:
-    # Only run on Pull Requests within the same repository, and not from forks as they don't work
-    if: github.event.pull_request.head.repo.full_name == github.repository
     name: calibreapp/image-actions
     runs-on: ubuntu-latest
+    # Only run on main repo as PRs can't be updated on forks
+    if: github.repository == 'HTTPArchive/almanac.httparchive.org'
+    env:
+      COMPRESS_ONLY: false
     steps:
-      - uses: actions/checkout@v2.3.2
-      - name: calibreapp/image-actions
-        uses: docker://calibreapp/github-image-actions
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Set COMPRESS_ONLY variable to true
+      # If it's not a pull request then run in compressOnly mode
+      if: github.event_name != 'pull_request'
+      run: |
+        echo "::set-env name=COMPRESS_ONLY::true"
+    - name: Checkout branch
+      uses: actions/checkout@v2.3.2
+    - name: calibreapp/image-actions
+      id: calibre
+      uses: calibreapp/image-actions@master
+      with:
+        githubToken: ${{ secrets.GITHUB_TOKEN }}
+        compressOnly: ${{ env.COMPRESS_ONLY }}
```

We can conditionally set the `compressOnly` parameter inline, so no need to initialize an environment variable `COMPRESS_ONLY` with the default value `false` and then setting it to true in a conditional step.

```suggestion
        compressOnly: ${{ github.event_name != 'pull_request' }}
```

---

Clever! Almost bordering on too clever at the cost of readability but think it's just this side of it so let's do it.

---

I would have not suggested it, but I though that too many things were done up front just to set this one variable. :-)